lightwood/api/data_source.py

```diff
@@ -260,6 +260,12 @@ def __getitem__(self, idx):
         if self.enable_cache:
             self.transformed_cache[idx] = sample
 
+        # Add offset in output_weights (output_features might be categorical _and_ numerical)
+        self.output_weights_offset = 0
+        for otype, oidxs in zip(self.out_types, self.out_indexes):
+            if otype != ColumnDataTypes.CATEGORICAL:
+                self.output_weights_offset += (oidxs[1] - oidxs[0])
```

Comment:
Actually, I now realize this does not work in the general case, so I'll be modifying it to comply with any ordering of any set of feature types